pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AntlrBaseNode.java

```diff
@@ -83,7 +83,16 @@ public void setUserData(final Object userData) {
 
     @Override
     public Node jjtGetChild(final int index) {
-        return (Node) children.get(index); // TODO: review if all children are Nodes
+        try {
+            return (Node) children.get(index);
+        } catch (final ClassCastException e) {
+            return new AntlrBaseNode();
+        }
+    }
+
+    @Override
+    public String getImage() {
```

Comment:
The image attribute is not supposed to be the text of the node. I'd rather keep it separate (the previous `@Text` attribute was fine). 

A practical reason for that is in the future, other languages may have a way to get the text of their node, in which case that wouldn't be fetched with `getImage`, for compatibility, but probably with a `getText` method, or `getSourceCode` like in #1728.

Also we *may* deprecate or otherwise put `getImage` out of use as soon as 7.0, given its contract is super loose.

Reply:
Ok, we can keep that as it was and use @Text attribute